rust/processor/src/worker.rs

```diff
@@ -298,44 +301,42 @@ impl Worker {
         // Create a gap detector task that will panic if there is a gap in the processing
         let (gap_detector_sender, gap_detector_receiver) =
             kanal::bounded_async::<ProcessingResult>(BUFFER_SIZE);
-        let (processor, gap_detection_batch_size, gap_detector_sender) =
-            if self.processor_config.is_parquet_processor() {
-                let processor = build_processor(
-                    &self.processor_config,
-                    self.per_table_chunk_sizes.clone(),
-                    self.deprecated_tables,
-                    self.db_pool.clone(),
-                    Some(gap_detector_sender.clone()),
-                );
-                let gap_detection_batch_size: u64 = self.parquet_gap_detection_batch_size;
+        let is_parquet_processor = self.processor_config.is_parquet_processor();
+        let (processor, gap_detection_batch_size) = if is_parquet_processor {
```

Comment:
Nit, you could just return `maybe_gap_detector_sender` and `gap_detection_batch_size` from this block and then only have `build_processor` in one place.